packages/transactions/src/new-signatures.ts

```diff
@@ -92,10 +92,15 @@ export async function newPartiallySignTransaction(
     });
 }
 
-export async function newSignTransaction(
+// Extract the difference between two types (any property in T but not in U)
+type Diff<T, U> = {
+    [P in Exclude<keyof T, keyof U>]: T[P];
+};
+
+export async function newSignTransaction<T extends NewTransaction>(
     keyPairs: CryptoKeyPair[],
-    transaction: NewTransaction,
-): Promise<FullySignedTransaction> {
+    transaction: T,
+): Promise<Diff<T, NewTransaction> & FullySignedTransaction> {
```

Comment:
Isn't it enough to do:

```suggestion
): Promise<T & FullySignedTransaction> {
```

since `T extends NewTransaction` anyway?

Reply:
I knew `Diff` felt too hacky lol, thanks! That's way better!